File: SplendoSharedApp/SplendoSharedApp/MainViewModel.m
//
//  MainViewModel.m
//  SplendoSharedApp
//
//  Created by Gijs van Veen on 27/01/2017.
//  Copyright © 2017 Gijs van Veen. All rights reserved.
//

#import "MainViewModel.h"
#import "MainModel.h"

@implementation MainViewModel

MainModel* model;

- (id) init {
    self = [super init];
    if (self != nil) {
        model = [MainModel new];
        [self updateLabelText];
    }
    return self;
}

- (void) updateLabelText {
    self.labelText = [NSString stringWithFormat:@"Current Count %d", [model counter]];
}

- (void) onButtonClicked {
    [model raiseCounter];
    [self updateLabelText];
}

@end

File: app/src/main/java/android/app/splendo/viewmodel/MainViewModel.java
package android.app.splendo.viewmodel;

import android.app.splendo.ObservableBuilder;
import android.app.splendo.SharedObservable;
import android.app.splendo.model.MainModel;

/**
 * Created by gijsvanveen on 26/01/2017.
 */
public class MainViewModel {

    public final SharedObservable<String> labelText;

    private MainModel model = new MainModel();

    public MainViewModel(ObservableBuilder observableBuilder) {
        labelText = observableBuilder.getStringObservable();

        updateLabelText();
    }

    private void updateLabelText() {
        labelText.set(String.format("Current Count: %1$d", model.getCounter()));
    }

    public void onButtonClicked() {
        model.raiseCounter();
        updateLabelText();
    }

}
